### ISSUE TITLE
Out-of-order effect

### ISSUE THREAD

**Issue by dolie** [preactjs/signals#457]
Hello 👋, 

I got this error, may you help me understand where the problem is ? 🙏

This error occurs when I click a button to trigger some async calls.
I don't see any cyclic dependencies... And don't really understand the error, and why it occurs...

## Error & Callstack :

<img width="276" alt="Capture d’écran 2023-11-29 à 17 58 48" src="https://github.com/preactjs/signals/assets/18349922/b6e5bdb8-dfe4-410e-9f27-ec8e76f55fcb">

## Files associated with callstack

<img width="331" alt="Capture d’écran 2023-11-29 à 17 50 04" src="https://github.com/preactjs/signals/assets/18349922/3e0bbb7a-3d63-4e70-b184-1ef8d50bfe30">  

<img width="376" alt="Capture d’écran 2023-11-29 à 18 00 04" src="https://github.com/preactjs/signals/assets/18349922/d7f05dd7-4ac0-4bdd-af0b-7a5013fccbd9">  

<img width="506" alt="Capture d’écran 2023-11-29 à 17 57 08" src="https://github.com/preactjs/signals/assets/18349922/9aefc063-88e2-4bed-bf99-4f5ecc97b21f">  

<img width="413" alt="Capture d’écran 2023-11-29 à 18 00 38" src="https://github.com/preactjs/signals/assets/18349922/567afb07-cfa6-49f9-9499-20fe0cfc1fbc">

**Comment by XantreGodlike** [comment]
In which environment are you using signals? Is it react or preact? Which bundler are u using? Vite? Webpack?
If it's react and vite - you can try out [my react integration](https://github.com/XantreGodlike/preact-signals/tree/main/packages/react) - I've tested it for this case (it happens while component recursive rerender itself)

**Comment by rschristian** [comment]
You need to provide an actual minimal reproduction, poorly cropped screenshots are far from adequate for helping us to help you.

**Comment by dolie** [comment]
Hello @XantreGodlike,
Thank you for your fast reply. 🚀
I resolved the error.

For posterity :

_Env : Preact & Vite & SWR_

I suspected that `useSignalEffects` were re-triggered by the `_mutate`.
So, I disabled all `useSignalEffect` (effects A,B & C) of page's components.
Effect `C` was so small and did not have much interactions with the two others. So I reactivate it. Keeps running ✅
`A` and `B` were both watching the mutated signal.
I decided to merge those into one `useSignalEffect`, and it corrected the all thing. ✅

useSignalEffect are powerful, but errors are difficult to track down... there were no clues between `_mutate` and those useSignalEffects... also "Out-of-order effect" is not really meaningful.

Also, if you know an eslint plugin that report cyclic dependencies in useSignalEffect, I am really interested. 🤓

Thank you again for your fast reply and for your work 🔥.
